webui/react/src/components/UPlotChart.tsx

```diff
@@ -44,10 +50,38 @@ const UPlotChart: React.FC<Props> = forwardRef((
   useEffect(() => {
     if (!chartDivRef.current || !hasData || !options) return;
 
+    const scaleMaxMin: Record<string, SerieMaxMin> = {};
+
     const optionsExtended = uPlot.assign(
       {
         cursor: { drag: { dist: 5, uni: 10, x: true, y: true } },
-        hooks: { ready: [ (chart: uPlot) => setChart(chart) ] },
+        hooks: {
+          ready: [ (chart: uPlot) => setChart(chart) ],
+          setScale: [ (uPlot: uPlot, scaleKey: string) => {
+            // need to ignore y because uPlot is adding margins to avoid having
+            // series line touching the max top of the chart.
+            if (scaleKey !== 'x') return;
+
+            const currentMax: number|undefined = uPlot.scales[scaleKey]?.max;
+            const currentMin: number|undefined = uPlot.scales[scaleKey]?.min;
+            const scaleSeries = uPlot.series.filter(serie => serie.scale === scaleKey);
+            let max: number|undefined = scaleMaxMin[scaleKey]?.max;
```

Comment:
Since we've already returned if scaleKey !== 'x', would it make sense to just use scaleMaxMin['x'] directly, or pull 'x' out as a constant?

Reply:
Technically this could work for Y axis too, so keeping this dynamic is fine.

The only thing stopping me from enabling this behaviour on Y axis is that uPlot is adding some "border" between the max value and the end of the chart. I'll try to understand that better and see if we can enable Y checks too.